packages/millicast-sdk/integration-tests/src/support-utils/generic.ts

```diff
@@ -0,0 +1,31 @@
+export async function retryUntilTrue(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
+    const timeout = timeoutSeconds * 1000
+    const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+    const startTime = new Date().getTime()
+    let result = false
+  
+    while (new Date().getTime() - startTime < timeout) {
+      try {
+        result = await func()
+        if (result) return result
+      } catch {}
+      await sleep(500)
+    }
+    return result
+  }
+  
+  export async function retryUntilFalse(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
+    const timeout = timeoutSeconds * 1000
+    const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+    const startTime = new Date().getTime()
+    let result = true
+  
+    while (new Date().getTime() - startTime < timeout) {
+      try {
+        result = await func()
+        if (result === false) return result
+      } catch {}
+      await sleep(500)
+    }
+    return result
+  }
```

Comment:
```suggestion
export async function retryUntilTrue(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
  return retryUntil(func, timeout, true);

  }
  
  export async function retryUntilFalse(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
  return retryUntil(func, timeout, false);
  }
  
  async function retryUntil(func: () => Promise<boolean>, timeoutSeconds = 10, condition: boolean = false): Promise<boolean> {
    const timeout = timeoutSeconds * 1000
    const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
    const startTime = new Date().getTime()
    let result = true

    while (new Date().getTime() - startTime < timeout) {
      try {
        result = await func()
        if (result) return result
      } catch {}
      await sleep(500)
    }
    return result
  }
  
```

Reply:
Nice, works ok but with adjustments:
- I have to put timeoutSeconds instead of timeout when calling retryUntil()
- 'if (condition === result) return result' in while loop
without it test failed or passed after first iteration. Please have a look @dubeyShivank 

```suggestion
export async function retryUntilTrue(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
  return retryUntil(func, timeoutSeconds, true);
  }
  
export async function retryUntilFalse(func: () => Promise<boolean>, timeoutSeconds = 10): Promise<boolean> {
  return retryUntil(func, timeoutSeconds, false);
  }
  
  async function retryUntil(func: () => Promise<boolean>, timeoutSeconds = 10, condition = false): Promise<boolean> {
    const timeout = timeoutSeconds * 1000
    const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
    const startTime = new Date().getTime()
    let result = true
    while (new Date().getTime() - startTime < timeout) {
      try {
        result = await func()
        if (condition === result) return result
      } catch {}
      await sleep(500)
    }
    return result
  }
```